testing/trino-tests/src/test/java/io/trino/execution/TestQueryTracker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.Session;
+import io.trino.connector.MockConnectorFactory;
+import io.trino.spi.Plugin;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.CountDownLatch;
+
+import static io.trino.SystemSessionProperties.QUERY_MAX_PLANNING_TIME;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+// Tests need to finish before strict timeouts. Any background work
+// may make them flaky
+@Test(singleThreaded = true)
+public class TestQueryTracker
+        extends AbstractTestQueryFramework
+{
+    private final CountDownLatch interrupted = new CountDownLatch(1);
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session defaultSession = testSessionBuilder()
+                .setCatalog("mock")
+                .setSchema("default")
+                .setSystemProperty(QUERY_MAX_PLANNING_TIME, "5s")
```

Comment:
We need to make this much tighter for this test or come up with a better way to test this.

Reply:
What exactly do you mean?
The feature is about making best effort to stop the planning, so ensuring it works in some cases should be enough IMO

Reply:
I'm saying that 5s is to long to sleep in a test.